apps/public-docsite-v9/src/Concepts/Theming.stories.mdx

```diff
@@ -0,0 +1,140 @@
+}
+```
+
+For more details, see [Styling components](?path=/docs/concepts-developer-styling-components--page).
+
+### Do not use CSS variables directly
+
+**⚠ Never use theme CSS variables directly!** The CSS variables implementation of the theme is internal to the library. We might eventually decide to change the variable names, hash them or even use direct values instead of some variables. Always use the `tokens` to access the theme.
+
+## Available themes
+
+Fluent UI currently exports following themes:
+
+- Web Light (`webLightTheme`)
+- Web Dark (`webDarkTheme`)
+- Web High Contrast (`webHighContrastTheme`)
+- Teams Light (`teamsLightTheme`)
+- Teams Dark (`teamsDarkTheme`)
+- Teams High Contrast (`teamsHighContrastTheme`)
+
+### High contrast themes
+
+⚠ Do not use High Contrast themes! All Fluent UI components support Windows High Contrast mode automatically regardless the used theme.
+Using Windows high contrast mode is recommended to all customers that are using Fluent UI.
+
+Hardcoded High Contrast themes are considered legacy, to be used only in applications which explicitly support those.
+
+## Customizing the theme
+
+There are multiple ways how an application can customize a theme.
+
+### Custom Brand ramp
+
+Brand ramp is a color ramp going from dark to light colors:
+
+<img src={require('../../public/brand-ramp-example.png')} alt="Example of a brand ramp" />
+
+A theme is a function of brand ramp. To use a theme with a custom brand ramp, instead of importing the pre-created theme, you can use theme factory functions.
+
+The following factory functions are available:
+
+- `createLightTheme()`
+- `createDarkTheme()`
+- `createTeamsDarkTheme()`
+- `createHighContrastTheme()`
+
+```tsx
+import { BrandVariants, createLightTheme, createDarkTheme } from '@fluentui/react-components';
+
+const customBrandRamp: BrandVariants = {
+  10: '#008',
+  //...
+  160: '#88F',
+};
+
+export const customLightTheme = createLightTheme(customBrandRamp);
+export const customDarkTheme = createDarkTheme(customBrandRamp);
+```
+
+### Overriding existing tokens
+
+If the existing tokens do not fulfill your needs, you should talk to your designer instead of overriding tokens.
+
+Theme is a flat object containing [token name; CSS value] pairs. You can copy the object and overwrite any tokens you wish.
+
+```tsx
+import { webLightTheme, Theme } from '@fluentui/react-components';
+
+export const customLightTheme: Theme = {
+  ...webLightTheme,
+  colorNeutralForeground1: '#555', // overriden token
+};
+```
+
+### Extending theme with new tokens
+
+Similarly to overriding existing tokens, you can add custom tokens as well.
+
+⚠ Components which use custom tokens cannot be shared among applications. Keep in mind that once a component which uses custom tokens will be shared among applications, all the applications must also add the custom tokens. Instead of adding custom tokens inside potentially reusable components, you should talk to design.
```

Comment:
```suggestion
⚠ Components which use custom tokens cannot be shared between applications. Keep in mind that any application which uses a component with custom tokens must also add the custom tokens to its own themes. Instead of adding custom tokens inside potentially reusable components, you should talk to design.
```